pkgs/tools/filesystems/fatresize/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, parted, substituteAll, utillinux, ... }:
+
+stdenv.mkDerivation rec {
+  
+  version = "1.0.2";
+  name = "fatresize-${version}";
+
+  src = fetchurl {
+    url = mirror://sourceforge/fatresize/fatresize-1.0.2.tar.bz2;
+    sha256 = "04wp48jpdvq4nn0dgbw5za07g842rnxlh9mig4mslz70zqs0izjm";
+  };
+  
+  phases = [
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "installPhase"
+  ];
+  
+  buildInputs = [ parted utillinux ];
+  
+  patches = [ ./parted_nix.patch ];
```

Comment:
This doesn't seem to add anything Nix-specific like the file name would imply. Could you elaborate a bit more (preferably in the patch header) where this patch comes from, what exactly it does and why?